### PR TITLE
fix: allow adding custom `Vary` to static assets handler

### PR DESCRIPTION
**Pull request by daniluk4000**
### 🔗 Linked issue

Closes https://github.com/nitrojs/nitro/issues/2834

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Allow to auto-extend Vary header if Accept-Encoding is passed from Nitro. 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/internal/static.ts [hunk 1]
```diff
@@ -49,7 +49,11 @@ export default eventHandler((event) => {
     "",
   ];
   if (encodings.length > 1) {
-    setResponseHeader(event, "Vary", "Accept-Encoding");
```
Comment (pi0) [inline src/runtime/internal/static.ts:52]: Maybe we can try `appendResponseHeader`?

Comment (daniluk4000) [reply src/runtime/internal/static.ts:52]: I _knew_ there was a better way but couldn't find it :D

image

Still curious why it works different all the time... 

Anyways, fixed.

**Comment by autofix-troubleshooter** [comment]
Hi! I'm the <a href="https://autofix.ci">autofix logo<strong>autofix.ci</strong></a> troubleshooter bot.

It looks like you correctly set up a CI job that uses the autofix.ci GitHub Action, but the [**autofix.ci GitHub App**](https://github.com/apps/autofix-ci/) has not been installed for this repository. This means that autofix.ci unfortunately does not have the permissions to fix this pull request. If you are the repository owner, please install the app and then restart the CI workflow! 😃

File: test/tests.ts [hunk 2]
```diff
@@ -310,6 +310,20 @@ export function testNitro(
     expect(data).toMatch("<h1 >Hello JSX!</h1>");
   });
 
+  it.skipIf(ctx.preset.startsWith('cloudflare') || ctx.preset === 'vercel-edge')(
```
Comment (pi0) [inline test/tests.ts:313]: Is the issue with this two diff in space? (we can normalize `headers['vary']` to remove space)

Comment (daniluk4000) [reply test/tests.ts:313]: Here I had something even different with Accept-Encoding not applying at all. I guess it depends on serve static param

Comment (pi0) [reply test/tests.ts:313]: Ah right.

**Comment by daniluk4000** [comment]
I sadly do not know locally what Prettier wants from me

**Comment by pi0** [comment]
Thnx!

**Pull request merged**